Source/monster.cpp

```diff
@@ -3025,16 +2967,12 @@ void MegaAi(int monsterId)
 		return;
 	}
 
-	int fx = monster.enemyPosition.x;
-	int fy = monster.enemyPosition.y;
-	mx = monster.position.tile.x - fx;
-	my = monster.position.tile.y - fy;
 	Direction md = GetDirection(monster.position.tile, monster.position.last);
 	if (monster.activeForTicks < UINT8_MAX)
 		MonstCheckDoors(monster);
 	int v = GenerateRnd(100);
-	int dist = std::max(abs(mx), abs(my));
-	if (dist >= 2 && monster.activeForTicks == UINT8_MAX && dTransVal[monster.position.tile.x][monster.position.tile.y] == dTransVal[fx][fy]) {
+	int dist = monster.longestDistanceToEnemy();

```

Comment:
move to before first usage at line 2961

Reply:
The original code was also calculating it twice, so I keep that behaviour.

Perhaps the call the SkeletonAi is updating the distance and it needs to be calculated again?

Reply:
Done